smack-experimental/src/main/java/org/jivesoftware/smackx/spoiler/element/SpoilerElement.java

```diff
@@ -0,0 +1,101 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.spoiler.element;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+import org.jivesoftware.smackx.spoiler.SpoilerManager;
+
+public class SpoilerElement implements ExtensionElement {
```

Comment:
The class's API is missing convenient methods to check whether or not an Message contains a spoiler extension element and to retrieve the metadata of an possibly existing spoiler extension element. For example something like
`static boolean containsSpoiler(Message)`
`static Map<String,String> getSpoilers(Message)`

I wonder if the API should make it possible to determine by a single `getSpoilers()` invocation if there was no spoiler, or if there was just a single empty spoiler. So `getSpoilers()` returns 'null' in case there are no spoilers, and the empty Map if there are just empty spoilers.

Reply:
Done